.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,211 @@
+          go-version: 1.18
+      - name: Build vcluster cli
+        run: |
+          set -x
+          GO111MODULE=on GOFLAGS=-mod=vendor go build -v -o ./vcluster ./cmd/vclusterctl/main.go
+      - name: Upload vcluster cli to artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: vcluster
+          path: ./vcluster
+          retention-days: 7
+
+  get-testsuites-dir:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - id: set-paths-matrix
+        run: |
+          set -x
+          sudo apt-get install -y jq
+          paths=$(ls -d ./test/e2e*)
+          echo "::set-output name=matrix::$(printf '%s\n' "${paths}" | jq -R . | jq -cs .)"
+    outputs:
+      matrix: ${{ steps.set-paths-matrix.outputs.matrix }}
+
+  e2e-tests:
+    name: Execute test suites
+    needs: [build-and-push-syncer-image, build-vcluster-cli, get-testsuites-dir]
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        distribution: ["k3s", "k8s", "k0s", "eks"]
+        test-suite-path: ${{fromJson(needs.get-testsuites-dir.outputs.matrix)}}
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+    
+    - name: Set up kind with K8s version v1.21.1
+      uses: engineerd/setup-kind@v0.5.0
+      with:
+        version: "v0.11.1"
+        image: kindest/node:v1.21.1
+    
+    - name: Testing kind cluster set-up
+      run: |
+          set -x
+          kubectl cluster-info
+          kubectl get pods -n kube-system
+          echo "kubectl config current-context:" $(kubectl config current-context)
+          echo "KUBECONFIG env var:" ${KUBECONFIG}
+    
+    - name: Download vcluster cli
+      uses: actions/download-artifact@v2
+      with:
+        name: vcluster
+    
+    - name: Download syncer image
+      uses: actions/download-artifact@v2
+      with:
+        name: vcluster_syncer
+
+    - name: Create vcluster
+      run: |
+        set -x 
+        
+        sudo apt-get install -y sed
+        
+        sed -i "s|REPLACE_IMAGE_NAME|${{ env.IMAGE_NAME }}|g" ${{ matrix.test-suite-path }}/../commonValues.yaml
+        
+        docker load < vcluster_syncer
+        
+        kind load docker-image ${{ env.IMAGE_NAME }}
+        
+        chmod +x vcluster && sudo mv vcluster /usr/bin
+        
+        vcluster create ${{ env.VCLUSTER_SUFFIX }} -n ${{ env.VCLUSTER_NAMESPACE }} \
+        --create-namespace \
+        --connect=false \
+        --distro=${{ matrix.distribution }} \
+        -f ${{ matrix.test-suite-path }}/../commonValues.yaml \
+        -f ${{ matrix.test-suite-path }}/values.yaml
+      continue-on-error: true
+
+    - name: Wait until vcluster is ready
+      id: wait-until-vcluster-is-ready
+      run: |
+        set -x
+        kubectl wait --for=condition=ready pod -l app=${{ env.VCLUSTER_SUFFIX }} -n ${{ env.VCLUSTER_NAMESPACE }} --timeout=600s
+      continue-on-error: true
```

Comment:
For this step, I would add a condition that it only executes if the previous one didn't fail.
And we should probably reduce the timeout, 3-5mins should be enough. I was probably the one that set it to 10mins initially :sweat_smile:, but now with many more jobs this is becoming potentially more costly.